test/tests/AelinUpFrontDeal.purchase.t.sol

```diff
@@ -1579,6 +1611,80 @@ contract AelinUpFrontDealPurchaseTest is Test, AelinTestUtils, IAelinUpFrontDeal
+
+        MockERC721(collection721_1).mint(user1, 1);
+        MockERC721(collection721_1).mint(user1, 2);
+        MockERC721(collection721_1).mint(user1, 3);
+
+        uint256 totalPoolShares;
+        uint256 poolSharesAmount;
+
+        // nft gating setup
+        AelinNftGating.NftPurchaseList[] memory nftPurchaseList = new AelinNftGating.NftPurchaseList[](1);
+        uint256[] memory tokenIdsArray = new uint256[](2);
+        // we get the allocation for each collection
+        (uint256 purchaseCollection1, , , , ) = AelinUpFrontDeal(dealAddressNftGating721IdRanges).getNftCollectionDetails(
+            address(collection721_1)
+        );
+
+        // checks pre-purchase
+        bool NftIdUsed;
+        bool hasNftList;
+        (NftIdUsed, hasNftList) = AelinUpFrontDeal(dealAddressNftGating721IdRanges).getNftGatingDetails(
+            address(collection721_1),
+            1
+        );
+        assertFalse(NftIdUsed);
+        assertTrue(hasNftList);
+        (NftIdUsed, hasNftList) = AelinUpFrontDeal(dealAddressNftGating721IdRanges).getNftGatingDetails(
+            address(collection721_1),
+            2
+        );
+        assertFalse(NftIdUsed);
+        assertTrue(hasNftList);
+
+        vm.startPrank(user1);
+
+        poolSharesAmount = (purchaseCollection1 * 10 ** underlyingTokenDecimals) / purchaseTokenPerDealToken;
+        totalPoolShares = poolSharesAmount;
+        vm.assume(poolSharesAmount > 0);
+        deal(address(purchaseToken), user1, type(uint256).max);
+        purchaseToken.approve(address(dealAddressNftGating721IdRanges), type(uint256).max);
+        tokenIdsArray[0] = 1;
+        tokenIdsArray[1] = 3;
+        nftPurchaseList[0].collectionAddress = address(collection721_1);
+        nftPurchaseList[0].tokenIds = tokenIdsArray;
+
+        //Second token Id out of range should fail
+        vm.expectRevert("tokenId not in range");
+        AelinUpFrontDeal(dealAddressNftGating721IdRanges).acceptDeal(nftPurchaseList, merkleDataEmpty, purchaseCollection1);
+
+        //Second token Id in range should succeed
+        tokenIdsArray[1] = 2;
+        vm.expectEmit(true, false, false, true);
+        emit AcceptDeal(user1, purchaseCollection1, purchaseCollection1, poolSharesAmount, poolSharesAmount);
+        AelinUpFrontDeal(dealAddressNftGating721IdRanges).acceptDeal(nftPurchaseList, merkleDataEmpty, purchaseCollection1);
+        vm.stopPrank();
+
+        //checks post-purchase
+        (NftIdUsed, hasNftList) = AelinUpFrontDeal(dealAddressNftGating721IdRanges).getNftGatingDetails(
+            address(collection721_1),
+            1
+        );
+        assertTrue(NftIdUsed);
+        assertTrue(hasNftList);
+        (NftIdUsed, hasNftList) = AelinUpFrontDeal(dealAddressNftGating721IdRanges).getNftGatingDetails(
+            address(collection721_1),
+            2
+        );
+        assertTrue(NftIdUsed);
+        assertTrue(hasNftList);
+    }
+
```

Comment:
can you maybe add a test where we can check if having multiple IDs in range (let's say 3) allows you to purchase `collectionPurchase * 3`?